apps/io-wallet-support-func/openapi.yaml

```diff
@@ -0,0 +1,220 @@
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/GetCurrentWalletInstanceStatusByFiscalCodeBody"
+      responses:
+        "200":
+          description: Wallet Instance status successfully retrieved
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/WalletData"
+        "404":
+          $ref: "#/components/responses/NotFound"
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+
+  /health:
+    get:
+      operationId: healthCheck
+      summary: Used to check if the function is healthy
+      responses:
+        "200":
+          description: The function is health
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  message:
+                    type: string
+        "500":
+          $ref: "#/components/responses/InternalServerError"
+
+components:
+  securitySchemes:
+    FunctionsKey:
+      type: apiKey
+      name: X-Functions-Key
+      in: header
+
+  responses:
+    Forbidden:
+      description: The server understands the request but refuses to authorize it
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ProblemJson"
+
+    BadRequest:
+      description: Validation error
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ProblemJson"
+
+    Unauthorized:
+      description: Unauthorized
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ProblemJson"
+
+    NotFound:
+      description: The specified resource was not found
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ProblemJson"
+
+    Conflict:
+      description: There has been a conflict
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ProblemJson"
+
+    UnprocessableContent:
+      description: Unprocessable Content
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ProblemJson"
+
+    InternalServerError:
+      description: Internal Server Error
+      content:
+        application/json:
+          schema:
+            $ref: "#/components/schemas/ProblemJson"
+
+  schemas:
+    WalletData:
+      description: Current wallet information
+      type: object
+      properties:
+        id:
+          type: string
+        is_revoked:
+          type: boolean
+        created_at:
+          $ref: "#/components/schemas/Timestamp"
+        revoked_at:
+          $ref: "#/components/schemas/Timestamp"
+        device_details:
+          $ref: "#/components/schemas/DeviceDetails"
+      required:
+        - id
+        - is_revoked
+        - created_at
```

Comment:
```suggestion
    WalletInstanceStatus:
      description: Current wallet instance status
      type: object
      properties:
        id:
          type: string
        is_revoked:
          type: boolean
        created_at:
          $ref: "#/components/schemas/Timestamp"
        revoked_at:
          $ref: "#/components/schemas/Timestamp"
        device_details:
          $ref: "#/components/schemas/DeviceDetails"
      required:
        - id
        - is_revoked
        - created_at
```

Reply:
If the function is `getCurrentWalletInstanceStatus` I think it is right to rename it, even if it seems that it **does not return only the status** but also other information related to the **Wallet Instance**. So what do you think about _renaming the function_ to something like `GetCurrentWalletInstance` (without status) and having it return the entire Wallet Instance as done here anyway?

Reply:
yes sure, I forgot to change the name. thank you